latex/properties.tex

```diff
@@ -0,0 +1,141 @@
+    \label{eq:2}
+    \forall l_{0},..,l_{n} \in \ledgerState, l_{0} =
+    \left(
+      \begin{array}{c}
+        \left\{
+        \genesisUTxO
+        \right\} \\
+        \emptyset\\
+        \emptyset\\
+      \end{array}
+    \right)  \\
+    \implies \forall 0 < i \leq n, ((\exists c \in \DCert, l_{i-1}
+    \trans{delegw}{c} l_{i}) \vee (\exists tx \in \Tx: l_{i-1} \trans{utxow}{tx}
+    l_{i}))\\ \implies \applyFun{validLedgerState}(l_{n})
+  \end{multline*}
+  \label{def:valid-ledger-state}
+\end{definition}
+
+Definition~\ref{def:valid-ledger-state} defines a valid ledger state reachable
+from the genesis state via valid UTxO, stake delegation or stake pool
+transactions. This gives a constructive rule how to reach a valid ledger state.
+
+\subsection{Ledger Properties}
+\label{sec:ledger-properties}
+
+The following properties state the desired features of updating a valid ledger
+state.
+
+\begin{property}[\textbf{Preserve Balance Modulo Fee}]
+  \begin{multline*}
+    \forall \var{l}, \var{l'} \in \ledgerState: \applyFun{validLedgerstate}{l}\\
+    \implies \forall \var{tx} \in \Tx, \var{l} \trans{utxo}{tx} \var{l'}
+    \implies \fun{balance}(\applyFun{getUTxO}{l}) =
+    \fun{balance}(\applyFun{getUTxO}{l'}) + \applyFun{txfee}{tx}
+  \end{multline*}
+  \label{prop:ledger-properties-1}
+\end{property}
+
+Property~\ref{prop:ledger-properties-1} states that for each valid ledger $l$,
+if a transaction $tx$ is added to the ledger via the state transition rule
+$utxow$ to the new ledger state $l'$, the balance of the UTxOs in $l$ equals the
+balance of the UTxOs in $l'$ minus the transaction fees.
+
+\begin{property}[\textbf{Preserve Balance Restricted to TxIns in Balance of
+    TxOuts}]
+  \begin{multline*}
+    \forall \var{l}, \var{l'} \in \ledgerState: \applyFun{validLedgerstate}{l}\\
+    \implies \forall \var{tx} \in \Tx, \var{l} \trans{utxo}{tx} \var{l'}
+    \implies \fun{balance}(\applyFun{txins}{tx} \restrictdom
+    \applyFun{getUTxO}{l}) = \fun{balance}(\applyFun{txouts}{tx}) +
+    \applyFun{txfee}{tx}
+  \end{multline*}
+  \label{prop:ledger-properties-2}
+\end{property}
+
+Property~\ref{prop:ledger-properties-2} states the more detailed relation of the
+balances change. For ledgers $l, l'$ and a transaction $tx$ as above, the
+balance of the UTxOs of $l$ restricted to those whose domain is in the set of
+transaction inputs of $tx$ equals the balance of the transaction outputs of $tx$
+minus the transaction fees.
+
+\begin{property}
+  \begin{multline*}
+    \forall \var{l}, \var{l'} \in \ledgerState: \applyFun{validLedgerstate}{l}\\
+    \implies \forall \var{tx} \in \Tx, \var{l} \trans{utxo}{tx} \var{l'}
+    \implies \forall \var{out} \in \applyFun{txouts}{tx},\\ \exists \var{utxo'}
+    \in \applyFun{getUTxo}{l'}, \exists \var{in} \in \applyFun{txins}{tx},
+    \var{in} \mapsto \var{out} = utxo' \wedge utxo' \not\in
+    \applyFun{getUTxO}{l}
+  \end{multline*}
+  \label{prop:ledger-properties-3}
+\end{property}
+
```

Comment:
txouts tx is of type UTxO, do you mean just the TxOuts of this UTxO? If so,
in |-> out is a single pair, do you mean (in |-> out) is in utxo'?